templates/internet-of-things/contact-us-hannover.html

```diff
@@ -0,0 +1,174 @@
+            {# These are honey pot fields to catch bots #}
+            <li class="u-off-screen">
+              <label class="website" for="website">Website:</label>
+              <input name="website" type="text" class="website" autocomplete="off" value="" id="website" tabindex="-1" />
+            </li>
+            <li class="u-off-screen">
+              <label class="name" for="name">Name:</label>
+              <input name="name" type="text" class="name" autocomplete="off" value="" id="name" tabindex="-1" />
+            </li>
+            {# End of honey pots #}
+            <li class="p-list__item">
+              <button type="submit" class="p-button--positive" onclick="dataLayer.push({'event' : 'GAEvent', 'eventCategory' : 'Form', 'eventAction' : 'hannover contact-us', 'eventLabel' : 'hannover', 'eventValue' : undefined });">Submit</button>
+            </li>
+          </ul>
+          <input type="hidden" aria-hidden="true" aria-label="hidden field" name="formid" value="4479" />
+          <input type="hidden" aria-hidden="true" aria-label="hidden field" name="returnURL" value="/internet-of-things/contact-us-hannover" />
+          <input type="hidden" aria-hidden="true" aria-label="hidden field" name="Consent_to_Processing__c" value="yes" />
+          <input type="hidden" aria-hidden="true" aria-label="hidden field" name="utm_campaign" id="utm_campaign" value="" />
+          <input type="hidden" aria-hidden="true" aria-label="hidden field" name="utm_medium" id="utm_medium" value="" />
+          <input type="hidden" aria-hidden="true" aria-label="hidden field" name="utm_source" id="utm_source" value="" />
+          <input type="hidden" aria-hidden="true" aria-label="hidden field" name="utm_content" id="utm_content" value="" />
+          <input type="hidden" aria-hidden="true" aria-label="hidden field" name="utm_term" id="utm_term" value="" />
+          <input type="hidden" aria-hidden="true" aria-label="hidden field" name="GCLID__c" id="GCLID__c" value="" />
+          <input type="hidden" aria-hidden="true" aria-label="hidden field" name="Facebook_Click_ID__c" id="Facebook_Click_ID__c" value="" />
+        </fieldset>
+      </form>
+    </div>
+  </div>
+</section>
+
+<script>
+  function createMessage() {
+    var message = "";
+    var formFields = document.querySelectorAll(".js-formfield");
+
+    formFields.forEach(function (formField) {
+      var comma = "";
+      var fieldTitle = formField.querySelector(".p-heading--5");
+      var inputs = formField.querySelectorAll("input");
+      if (fieldTitle) {
+      message += fieldTitle.innerText + "\r\n";
+      }
+      
+      inputs.forEach(function (input) {
+        if (input.checked) {
+          message += comma + input.value + "\r\n\r\n";
+          comma = ", ";
+        }
+      });
+    });
+
+    return message;
+  }
+
+  function handleSubmission(e) {
+    e.preventDefault();
+    var leadComments = document.getElementById("Comments_from_lead__c");
```

Comment:
What is the reasoning behind using 'querySelector' over 'getElementById'?

Reply:
querySelector allows you to search within an element instead of a global document search. It's safer to just operate within the scope of a single form.